.github/workflows/build-cmake.yaml

```diff
@@ -0,0 +1,176 @@
+name: CMake
+
+on:
+  workflow_dispatch: # Manual trigger
+  push:
+    branches:
+      - master
+  pull_request:
+    paths-ignore:
+      - .gitignore
+      - .github/**
+      - '!.github/workflows/build-cmake.yaml'  # Run check on self change
+      - CONTRIBUTORS
+      - LICENSE
+      - NOTICE
+      - README.md
+      - android/**
+      - iphone/**
+      - data/strings/**
+      - docs/**
+      - packaging/**
+      - pyhelpers/**
+      - tools/**
+      - '!tools/python/test_server/**'
+      - xcode/**
+
+# Cancels previous jobs if the same branch or PR was updated again.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        cmake-build-type: [ Debug, Release ]
+        environment:
+          - os: ubuntu-24.04
+            cc: clang-18
+            cxx: clang++-18
+            no-unity: ON
+            run-tests: OFF
+
+          - os: ubuntu-24.04
+            cc: clang-18
+            cxx: clang++-18
+
+          - os: ubuntu-24.04
+            cc: gcc-14
+            cxx: g++-14
+
+          - os: macos-15
+            cc: clang
+            cxx: clang++
+            developer-dir: /Applications/Xcode_16.app/Contents/Developer
```

Comment:
nit: move it to the env of the macOS job below, where other env vars are already defined.